### PR TITLE
[Virt] add sidecar test

### PR DESCRIPTION
**Pull request by ILpinto**
Add sidecar test.

### CODE REVIEW

File: resources/resource.py [hunk 1]
```diff
@@ -214,7 +215,8 @@ def _extract_data_from_yaml(self, yaml_data):
         Args:
             yaml_data (dict): Dict from yaml file
         """
-        self.namespace = yaml_data.get('metadata').get('namespace')
+        namespace = yaml_data.get('metadata').get('namespace')
+        self.namespace = self.namespace if not namespace else namespace
```
Comment (myakove) [inline resources/resource.py]: self.namespace = namespace or self.namespace

File: tests/virt/config.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from tests.config import *  # noqa: F401
+
+# VIRT namespace
+VIRT_NS = "cnv-virt-ns"
+
+#COMMANDS
+CHECK_DMIDECODE_PACKAGE = "sudo dmidecode -s baseboard-manufacturer | grep 'Radical Edward' | wc -l\n"
```
Comment (myakove) [inline tests/virt/config.py]: Do you need this in the global virt config?

File: tests/virt/fixtures.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import logging
+
+import pytest
+from tests.test_utils import wait_for_vm_interfaces
+from resources.virtual_machine_instance import VirtualMachineInstance
+from utilities import types
+from . import config
+
+LOGGER = logging.getLogger(__name__)
```
Comment (myakove) [inline tests/virt/fixtures.py]: LOGGER is not used.

File: tests/virt/fixtures.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+def create_vmi_with_yaml(request):
+    """
+    create VMI and wait till it running with yaml (don't start VM unless it has running state True)
+    """
+    vm_name = request.cls.vm_name
+    vm_yaml = request.cls.vm_yaml
+    vmi = VirtualMachineInstance(name=vm_name, namespace=config.VIRT_NS)
+    
+    def fin():
+        assert vmi.delete(wait=True), "Failed to delete vm %vm_name".format(vm_name=vm_name)
```
Comment (myakove) [inline tests/virt/fixtures.py]: I don't think this is going to work, %vm_name :)

File: tests/virt/vm_with_sidecar/test_vm_with_sidecar.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+"""
+import logging
+import pytest
+
+from utilities import console
+from tests.virt import config
+from tests.virt.fixtures import create_vmi_with_yaml  # noqa: F401
+
+
+LOGGER = logging.getLogger(__name__)
```
Comment (myakove) [inline tests/virt/vm_with_sidecar/test_vm_with_sidecar.py]: LOGGER is not used.

File: tests/virt/vm_with_sidecar/test_vm_with_sidecar.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+"""
+VM with sidecar
+"""
+import logging
+import pytest
+
+from utilities import console
+from tests.virt import config
+from tests.virt.fixtures import create_vmi_with_yaml  # noqa: F401
```
Comment (myakove) [inline tests/virt/vm_with_sidecar/test_vm_with_sidecar.py]: Why did you added # noqa: F401?

Comment (ILpinto) [reply tests/virt/vm_with_sidecar/test_vm_with_sidecar.py]: removed

File: tests/virt/vm_with_sidecar/test_vm_with_sidecar.py (hunk 5 above)
Comment (ILpinto) [reply tests/virt/vm_with_sidecar/test_vm_with_sidecar.py]: removed

File: tests/virt/fixtures.py (hunk 4 above)
Comment (ILpinto) [reply tests/virt/fixtures.py]: Fixed

File: resources/resource.py (hunk 1 above)
Comment (ILpinto) [reply resources/resource.py]: Fixed

Comment (ILpinto) [reply resources/resource.py]: Fixed

File: tests/virt/config.py (hunk 2 above)
Comment (ILpinto) [reply tests/virt/config.py]: Moved to test

File: tests/virt/fixtures.py (hunk 3 above)
Comment (ILpinto) [reply tests/virt/fixtures.py]: Removed

File: tests/virt/fixtures.py [hunk 7]
```diff
@@ -19,9 +15,9 @@ def create_vmi_with_yaml(request):
     vmi = VirtualMachineInstance(name=vm_name, namespace=config.VIRT_NS)
     
     def fin():
-        assert vmi.delete(wait=True), "Failed to delete vm %vm_name".format(vm_name=vm_name)
+        assert vmi.delete(wait=True), "Failed to delete vm {vm_name}".format(vm_name=vm_name)
```
Comment (myakove) [inline tests/virt/fixtures.py]: f"Failed to delete vm {vm_name}"  :)

Comment (myakove) [reply tests/virt/fixtures.py]: Also create and delete will generate error logs when it fail.

File: tests/virt/vm_with_sidecar/test_vm_with_sidecar.py [hunk 8]
```diff
@@ -3,15 +3,13 @@
...
 import pytest
 
 from utilities import console
 from tests.virt import config
-from tests.virt.fixtures import create_vmi_with_yaml  # noqa: F401
+from tests.virt.fixtures import create_vmi_with_yaml
 
-
-LOGGER = logging.getLogger(__name__)
+CHECK_DMIDECODE_PACKAGE = "sudo dmidecode -s baseboard-manufacturer | grep 'Radical Edward' | wc -l\n"
```
Comment (myakove) [inline tests/virt/vm_with_sidecar/test_vm_with_sidecar.py]: If it used only once just use it in the call.
vm_console.sendline("sudo dmidecode -s baseboard-manufacturer | grep 'Radical Edward' | wc -l\n")

Comment (ILpinto) [reply tests/virt/vm_with_sidecar/test_vm_with_sidecar.py]: Better to use Variable the strings more readable.

File: tests/virt/fixtures.py (hunk 7 above)
Comment (ILpinto) [reply tests/virt/fixtures.py]: if the generate log will report in failure (assert) better to remote the log.

**Pull request merged**